utilities/transactions/pessimistic_transaction_db.cc

```diff
@@ -549,11 +578,7 @@ bool WritePreparedTxnDB::IsInSnapshot(uint64_t prep_seq,
   CommitEntry64b dont_care;
   CommitEntry cached;
   bool exist = GetCommitEntry(indexed_seq, &dont_care, &cached);
-  if (!exist) {
-    // It is not committed, so it must be still prepared
-    return false;
-  }
-  if (prep_seq == cached.prep_seq) {
+  if (exist && prep_seq == cached.prep_seq) {
```

Comment:
Then it is not found in the commit map and we move on with the rest of the logic.
The bug that is fixed by this change was that previously we assumed that if it is non-existent it means that it is never committed (otherwise it would be either existent with the same prep_seq or be replaced). However in recovery we do not populate the commit map with the previously committed txns and non-existent cannot indicate never-committed.

Reply:
I see. Can you add a small comment here for the non-exist case?

Reply:
Sure. Will do.